azurerm/internal/services/network/network_packet_capture_migration_resource.go

```diff
@@ -0,0 +1,36 @@
+package network
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func ResourceNetworkPacketCaptureMigrateState(v int, is *terraform.InstanceState, _ interface{}) (*terraform.InstanceState, error) {
+	switch v {
+	case 0:
+		log.Println("[INFO] Found AzureRM Network Packet Capture State v0; migrating to v1")
+		return resourceNetworkPacketCaptureStateV0toV1(is)
+	default:
+		return is, fmt.Errorf("Unexpected schema version: %d", v)
+	}
+}
+
+func resourceNetworkPacketCaptureStateV0toV1(is *terraform.InstanceState) (*terraform.InstanceState, error) {
+	if is.Empty() {
+		log.Println("[DEBUG] Empty InstanceState; nothing to migrate.")
+		return is, nil
+	}
+
+	log.Printf("[DEBUG] ARM Network Packet Capture before Migration: %#v", is.Attributes)
+
+	newID := strings.Replace(is.Attributes["id"], "/NetworkPacketCaptures/", "/packetCaptures/", 1)
```

Comment:
Hi @neil-yechenwei - Can you post the manual test results / output here? (Ideally with an `apply` rather than a `plan` as this will actually update the state and exercise the code.).

Thanks

Reply:
@jackofallops , sure.
1. Run `tf apply` to create Network Packet Capture with provider.azurerm v2.0.0
![image](https://user-images.githubusercontent.com/19754191/92844010-f7615e00-f417-11ea-9605-8d69ce6429d0.png)
2. Switch provider.azurerm from v2.0.0 to this branch
3. Run `tf plan`
![image](https://user-images.githubusercontent.com/19754191/92844439-6e96f200-f418-11ea-82ff-23be6b58ad2c.png)
4. Run `tf apply`
![image](https://user-images.githubusercontent.com/19754191/92844555-91290b00-f418-11ea-8051-66c5de5595f6.png)
5. Update the name of network packet capture in tfconfig
6. Run `tf apply`
![image](https://user-images.githubusercontent.com/19754191/92845403-87ec6e00-f419-11ea-80a1-3c41bddbe69c.png)
7. Run `tf plan`
![image](https://user-images.githubusercontent.com/19754191/92845491-9f2b5b80-f419-11ea-8fee-d9dc18751f5a.png)
8. Run `tf destroy`
![image](https://user-images.githubusercontent.com/19754191/92846376-9a1adc00-f41a-11ea-820a-374067297123.png)